File: csharp/pascals-triangle/PascalsTriangle.cs
using System;
using System.Collections.Generic;

public static class PascalsTriangle
{
    public static IEnumerable<IEnumerable<int>> Calculate(int rows)
    {
        int[][] pascalsTriangle = new int[rows][];
        if (rows == 0) return pascalsTriangle;

        pascalsTriangle[0] = new int[]{1};
        for (int row = 1; row < rows; row++)
        {
            int[] currentRow = new int[row + 1];
            currentRow[0] = 1;
            currentRow[currentRow.Length - 1] = 1;

            for (int column = 1; column < currentRow.Length - 1; column++)
            {
                currentRow[column] = pascalsTriangle[row - 1][column - 1] + pascalsTriangle[row - 1][column];
            }
            pascalsTriangle[row] = currentRow;
        }
        return pascalsTriangle;
    }
}

File: javascript/pascals-triangle/pascals-triangle.js
//
// This is only a SKELETON file for the 'Pascals Triangle' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export class Triangle {
  constructor(numberOfRows) {
    this._rows = this.createRows(numberOfRows);
  }

  get lastRow() {
    return this._rows[this._rows.length - 1];
  }
  
  get rows() {
    return this._rows;
  }

  createRows(numberOfRows) {
    const rows = [];
    for (let row = 0; row < numberOfRows; row++) {
      let currentRow = [];
      for (let column = 0; column < row + 1; column++) {
        if (column === 0 || column === row) currentRow.push(1);
        else currentRow.push(rows[row - 1][column - 1] + rows[row - 1][column]);
      }
      rows.push(currentRow)
    }
    return rows;
  }
}
